chat-android/src/main/java/com/ably/chat/RoomLifecycleManager.kt

```diff
@@ -1,116 +1,48 @@
 package com.ably.chat
 
 import com.ably.annotations.InternalAPI
+import io.ably.lib.realtime.Channel
 import io.ably.lib.realtime.ChannelState
+import io.ably.lib.realtime.ChannelStateListener.ChannelStateChange
 import io.ably.lib.types.AblyException
-import io.ably.lib.types.ErrorInfo
-import kotlin.coroutines.resume
-import kotlin.coroutines.resumeWithException
 import kotlinx.coroutines.CoroutineScope
-import kotlinx.coroutines.async
-import kotlinx.coroutines.awaitAll
-import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.Job
 import kotlinx.coroutines.delay
-import kotlinx.coroutines.suspendCancellableCoroutine
+import kotlinx.coroutines.launch
 
 /**
- * An interface for features that contribute to the room status.
+ * An interface for features that contribute to the room.
  */
-internal interface ContributesToRoomLifecycle : EmitsDiscontinuities, HandlesDiscontinuity {
-
+internal interface ContributesToRoomLifecycle {
```

Comment:
Should we rename to `ChatFeature` or something similar, I believe `ContributesToRoomLifecycle` no longer fits very good

Reply:
https://github.com/ably/ably-chat-kotlin/pull/124/commits/f2ee2b19ef6964ea168e1c8707f5c9b30b5c2341